examples/1-hello-world/README.md

```diff
@@ -4,131 +4,283 @@ This is very simple `bin` crate generated by `cargo new`, with no additional
+should build this project for. This is _not_ a [Rust target triple], but rather
+a [Nix system parameter] such as `"x86_64-linux"` or `"x86_64-darwin"`. By
+default, we want this string to be set to `builtins.currentSystem` if left
+unspecified by the user, so we use the `?` operator to set a default.
+
+[Rust target triple]: https://forge.rust-lang.org/release/platform-support.html
+[Nix system parameter]: https://nixos.org/nixos/nix-pills/nixpkgs-parameters.html#idm140737319760528
+
+The second argument, `nixpkgsMozilla`, fetches [nixpkgs-mozilla] from GitHub.
+This repository contains a [handy Nixpkgs overlay for Rust] which allows you to
+fetch a version of Rust using [release channels] similar to Rustup, e.g.
+`"stable"`, `"nightly"`, or `"1.41.0"`.
+
+[nixpkgs-mozilla]: https://github.com/mozilla/nixpkgs-mozilla
+[handy Nixpkgs overlay for Rust]: https://github.com/mozilla/nixpkgs-mozilla#rust-overlay
+[release channels]: https://doc.rust-lang.org/edition-guide/rust-2018/rustup-for-managing-rust-versions.html
+
+The final argument, `cargo2nix`, fetches the [cargo2nix] repository from GitHub.
+To build our project, we need to fetch a fixed release of `cargo2nix` so we can
+use the provided Nixpkgs overlay from it to build our Rust project. By default,
+we fetch from the Git tag `v0.8.0`.
+
+Next, we need to write the body of our function. To do this, we declare a `let`
+block and import [Nixpkgs] using our three function arguments from earlier:
+
+[cargo2nix]: https://github.com/tenx-tech/cargo2nix
+[Nixpkgs]: https://github.com/NixOS/nixpkgs
+
+```nix
+let
+  rustOverlay = import "${nixpkgsMozilla}/rust-overlay.nix";
+  cargo2nixOverlay = import "${cargo2nix}/overlay";
+
+  pkgs = import <nixpkgs> {
+    inherit system;
+    overlays = [ rustOverlay cargo2nixOverlay ];
+  };
+in
+  # TODO
+```
 
-Naturally, since this Cargo project contains no tests, the resulting test binary
-contains 0 tests.
+Now that Nixpkgs is imported with our two overlays, we now have a few extra
+functions available to us, courtesy of the `cargo2nix` overlay. Specifically, we
+have the entry point to `cargo2nix` called `rustBuilder`. Inside of
+`rustBuilder`, we have a function called `makePackageSet'` (note the single
+quote in the name) which can be used to build our project's `Cargo.nix` file.
 
-### Examining the test binaries
+To build our `hello-world` project, we change our `let` block from earlier to
+look like this:
 
-The `nix-build -A ci` invocation shown above produces a `result` symlink with a
-slightly different directory structure:
+```nix
+let
+  rustOverlay = import "${nixpkgsMozilla}/rust-overlay.nix";
+  cargo2nixOverlay = import "${cargo2nix}/overlay";
+
+  pkgs = import <nixpkgs> {
+    inherit system;
+    overlays = [ rustOverlay cargo2nixOverlay ];
+  };
+
+  rustPkgs = pkgs.rustBuilder.makePackageSet' {
+    rustChannel = "stable";
+    packageFun = import ./Cargo.nix;
+    packageOverrides = pkgs: pkgs.rustBuilder.overrides.all;
```

Comment:
Same here. Should we omit this line and reintroduce it in the second example?

Reply:
Honestly, I'm not sure. On one hand, we could remove it since it's technically optional and everything will still build without it.

But on the other hand, if we choose to remove it, I think it should be made clear that it will default to `pkgs.rustBuilder.overrides.all` regardless of whether it's specified or not, and we don't need to add it in order to get `reqwest` to compile. If it's present from the very beginning, then the user knows we didn't need to do anything special in order to get their system dependencies built.

Reply:
I don't know, what do you think @trha? 😄

Reply:
The first thing to get across in a hello world is what you absolutely can't run without.  Recommend omitting where possible

Reply:
Okay, that's fair! Maybe I could remove it from `hello-world`, but perhaps leave it in `bigger-project` but commented out, with a note saying "this is where the crate overrides come from."

Reply:
@antigravityla Okay, that's fair! Maybe I could remove it from `hello-world`, but perhaps leave it in `bigger-project` but commented out, with a note saying "this is where the crate overrides come from?"

Reply:
Done in d214c36! I decided to remove it from `hello-world` but introduce it commented out in `bigger-project`.